src/golang/lib/lib_utils/utils.go

```diff
@@ -146,3 +146,30 @@ func ParseEmailConfig(conf auth.Config) (*shared.EmailConfig, error) {
 		Targets:  targets,
 	}, nil
 }
+
+func ParseSlackConfig(conf auth.Config) (*shared.SlackConfig, error) {
+	data, err := conf.Marshal()
+	if err != nil {
+		return nil, err
+	}
+
+	var c struct {
```

Comment:
nit: there's not really a need to use yaml tags unless we are parsing from a yaml file, which we aren't doing in most of these use cases. we can clean up the rest of the codebase later, but just something worth leaving out going forward